src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFChart.java

```diff
@@ -458,21 +492,50 @@ public XDDFDateAxis createDateAxis(AxisPosition pos) {
         return dateAxis;
     }
 
+    /**
+     * this method will return specified chart data with category and series values
+     *
+     * @param type chart type
+     * @param category category values of chart
+     * @param values series values of chart
+     * @return specified chart data.
+     */
     public XDDFChartData createData(ChartTypes type, XDDFChartAxis category, XDDFValueAxis values) {
-        Map<Long, XDDFChartAxis> categories = Collections.singletonMap(category.getId(), category);
-        Map<Long, XDDFValueAxis> mapValues = Collections.singletonMap(values.getId(), values);
+        Map<Long, XDDFChartAxis> categories = null;
+        Map<Long, XDDFValueAxis> mapValues = null;
+        
+        if(ChartTypes.PIE != type && ChartTypes.PIE3D != type)
```

Comment:
handled null pointer exception in case of pie chart.

Reply:
nice catch!

Reply:
thank you :)